src/build/tenscript/pest_parser.rs

```diff
@@ -0,0 +1,184 @@
+                    }
+                }
+            }
+            _ => unreachable!("shape phase")
+        }
+    }
+    Ok(shape_phase)
+}
+
+fn build(build_phase_pair: Pair<Rule>) -> Result<BuildPhase, ParseError> {
+    let mut phase = BuildPhase::default();
+    for pair in build_phase_pair.into_inner() {
+        match pair.as_rule() {
+            Rule::seed => {
+                phase.seed =
+                    match pair.into_inner().next().unwrap().as_str() {
+                        ":left-right" => Seed::LeftRight,
+                        ":right-left" => Seed::RightLeft,
+                        ":left" => Seed::Left,
+                        ":right" => Seed::Right,
+                        _ => unreachable!()
+                    };
+            }
+            Rule::node => {
+                phase.root = Some(node(pair).unwrap());
+            }
+            _ => unreachable!("build phase: {:?}", pair.as_rule()),
+        }
+    }
+    Ok(phase)
+}
+
+fn node(node_pair: Pair<Rule>) -> Result<TenscriptNode, ParseError> {
+    let pair = node_pair.into_inner().next().unwrap();
+    match pair.as_rule() {
+        Rule::face => {
+            let mut inner_pairs = pair.into_inner();
+            let face_name_string = inner_pairs.next().unwrap().as_str();
+            let face_name: FaceName = face_name_string[1..].try_into().unwrap();
+            let node = node(inner_pairs.next().unwrap()).unwrap();
+            Ok(TenscriptNode::Face {
+                face_name,
+                node: Box::new(node),
+            })
+        }
+        Rule::grow => {
+            let mut inner = pair.into_inner();
+            let forward_string = inner.next().unwrap().as_str();
+            let forward = match forward_string.parse() {
+                Ok(count) => { "X".repeat(count) }
+                Err(_) => { forward_string[1..forward_string.len() - 1].into() }
+            };
+            let scale_factor = scale(inner.next());
+            let post_growth_node = inner.next()
+                .map(|post_growth| Box::new(node(post_growth).unwrap()));
+            Ok(TenscriptNode::Grow {
+                forward,
+                scale_factor,
+                post_growth_node,
+            })
+        }
+        Rule::mark => {
+            let mark_name = pair.into_inner().next().unwrap().as_str();
+            Ok(TenscriptNode::Mark { mark_name: mark_name[1..].into() })
+        }
+        Rule::branch => {
+            Ok(TenscriptNode::Branch {
+                face_nodes: pair.into_inner()
+                    .map(|face_node| node(face_node).unwrap())
+                    .collect()
+            })
+        }
+        _ => unreachable!("node"),
+    }
+}
+
+fn scale(scale_pair: Option<Pair<Rule>>) -> f32 {
+    match scale_pair {
+        None => 1.0,
+        Some(scale_pair) => {
+            let scale_string = scale_pair.into_inner().next().unwrap().as_str();
+            scale_string.parse().unwrap()
+        }
```

Comment:
```suggestion
                Some(scale_pair) =>
            scale_pair
                .into_inner()
                .next()
                .unwrap()
                .as_str()
                .parse()
                .unwrap(),
```
Matter of preference